editor/src/components/canvas/ui-jsx-canvas-renderer/ui-jsx-canvas.spec.browser2.tsx

```diff
@@ -0,0 +1,316 @@
+  return `import * as React from "react";
+import { Scene, Storyboard, jsx } from "utopia-api";
+
+export var App = (props) => {
+  return (
+    <div
+      data-uid="other-app-root"
+      style={{
+        width: "100%",
+        height: "100%",
+        backgroundColor: "#FFFFFF",
+        position: "relative",
+      }}
+    >
+      <div
+        data-uid='div-red'
+        data-testid='div-red'
+        style={{
+          ${positionLine}
+          left: 50,
+          top: ${50 + elementTopShifts[0]},
+          width: 100,
+          height: 100,
+          backgroundColor: 'red',
+        }}
+        data-label='Red'
+      />
+      <div
+        data-uid='div-green'
+        data-testid='div-green'
+        style={{
+          ${positionLine}
+          left: 60,
+          top: ${60 + elementTopShifts[1]},
+          width: 100,
+          height: 100,
+          backgroundColor: 'green',
+        }}
+        data-label='Green'
+      />
+      <div
+        data-uid='div-blue'
+        data-testid='div-blue'
+        style={{
+          ${positionLine}
+          left: 70,
+          top: ${70 + elementTopShifts[2]},
+          width: 100,
+          height: 100,
+          backgroundColor: 'blue',
+        }}
+        data-label='Blue'
+      />
+    </div>
+  );
+};
+
+export var storyboard = (
+  <Storyboard data-uid="storyboard">
+    <Scene
+      data-uid="scene-1"
+      style={{ position: "absolute", left: 0, top: 0, width: 375, height: 812 }}
+    >
+      <App data-uid="app" />
+    </Scene>
+  </Storyboard>
+);
+`
+}
+
+function createAndRenderModifiedProject(modifiedFiles: {
+  [filename: string]: string
+}): Promise<EditorRenderResult> {
+  const project = createModifiedProject(modifiedFiles)
+  return renderTestEditorWithModel(project, 'await-first-dom-report')
+}
+
+function createExampleProject(
+  positionSetting: string | null,
+  elementTopShifts: [number, number, number],
+): Promise<EditorRenderResult> {
+  return createAndRenderModifiedProject({
+    [StoryboardFilePath]: exampleProject(positionSetting, elementTopShifts),
+  })
+}
+
+async function checkOverlappingElements(
+  positionSetting: string | null,
+  elementTopShifts: [number, number, number],
+  focusedOn: ElementPath | null,
+  targetDivId: string,
+  cmdPressed: boolean,
+  x: number,
+  y: number,
+  expectedSelectedElements: Array<ElementPath>,
+): Promise<void> {
+  const renderResult = await createExampleProject(positionSetting, elementTopShifts)
+
+  const targetDiv = renderResult.renderedDOM.getByTestId(targetDivId)
+  const targetDivBounds = targetDiv.getBoundingClientRect()
+
+  const canvasControlsLayer = renderResult.renderedDOM.getByTestId(CanvasControlsContainerID)
+
+  if (focusedOn != null) {
+    await act(async () => {
+      const dispatchDone = renderResult.getDispatchFollowUpActionsFinished()
+      await renderResult.dispatch([setFocusedElement(focusedOn)], true)
+      await dispatchDone
+    })
```

Comment:
`renderResult.dispatch` will internally call an `await act(() => await renderResult.dispatch(...`, so just awaiting `await renderResult.dispatch` is equivalent to getting the dispatchDone and awaiting that.

so I believe the above 4 lines are equivalent to this:
```suggestion
    await renderResult.dispatch([setFocusedElement(focusedOn)], true)
```